django/test/testcases.py

```diff
@@ -887,6 +887,57 @@ def assertXMLNotEqual(self, xml1, xml2, msg=None):
                 standardMsg = '%s == %s' % (safe_repr(xml1, True), safe_repr(xml2, True))
                 self.fail(self._formatMessage(msg, standardMsg))
 
+    def assertFormValid(self, form, msg=None):
+        """
+        Asserts that a form is valid
+        """
+        if not form.is_valid():
+            standardMsg = 'Form validation failed. Error(s):\n'
+            for field, field_errors in form.errors.items():
+                standardMsg += f'- {field}:\n'
+                for error in field_errors:
+                    standardMsg += f'    - {error}\n'
+            self.fail(self._formatMessage(msg, standardMsg))
+
+    def assertFormNotValid(self, form, expected_errors=None,
+                           allow_all_errors=True, msg=None):
+        """
+        Asserts that a form is not valid
+        """
+        error_list = []
+        if expected_errors:
+            for field, errors in expected_errors.items():
+                if isinstance(errors, tuple):
+                    for error in errors:
+                        error_list.append((field, error))
+                else:
+                    error_list.append((field, errors))
+
+        to_report = ''
+        if form.is_valid():
+            to_report = 'Form is valid\n'
```

Comment:
We don't need this. Without the `\n` it looks like this. Will have a look at other areas to see if we can put `\n` at the start instead of the end of lines. 

```
AssertionError: FormNotValid Error:
- Form is valid

During handling of the above exception, another exception occurred:
```

Reply:
Also, I don't think we should collect this in `to_report`. We can fail immediately with `self.fail('Form is valid')`.

Reply:
I completely agree, I already made this change.